pkg/controller/sparkapplication/sparkui.go

```diff
@@ -137,17 +159,30 @@ func createSparkUIService(
 	return &SparkService{
 		serviceName: service.Name,
 		servicePort: service.Spec.Ports[0].Port,
+		targetPort:  service.Spec.Ports[0].TargetPort,
 		serviceIP:   service.Spec.ClusterIP,
 	}, nil
 }
 
 // getWebUITargetPort attempts to get the Spark web UI port from configuration property spark.ui.port
 // in Spec.SparkConf if it is present, otherwise the default port is returned.
 // Note that we don't attempt to get the port from Spec.SparkConfigMap.
-func getUITargetPort(app *v1beta2.SparkApplication) string {
+func getUITargetPort(app *v1beta2.SparkApplication) (int, error) {
 	port, ok := app.Spec.SparkConf[sparkUIPortConfigurationKey]
 	if ok {
-		return port
+		return strconv.Atoi(port)
+	}
+	return strconv.Atoi(defaultSparkWebUIPort)
```

Comment:
We can change `defaultSparkWebUIPort` to an `int32`.

Reply:
done, I also changed the return type of `getUIServicePort` and `getUITargetPort` to int32